### PR TITLE
Expand solana-sdk API docs.

### PR DESCRIPTION
**Pull request by brson**
#### Summary of Changes

This adds crate docs to solana-sdk and one-line descriptions of most top level items.

This is mostly to improve the presentation of the docs, not as detailed as solana-program. A lot is just small changes for consistency with solana-program.

Fixes #

### CODE REVIEW

File: sdk/src/derivation_path.rs [hunk 1]
```diff
@@ -1,3 +1,13 @@
+//! Definitions of standard Solana [BIP-44] key derivation paths.
```
Comment (CriesofCarrots) [inline sdk/src/derivation_path.rs:1]: This module also handles non-Solana BIP-44 derivation paths. Should we allude to that in some fashion?

File: sdk/src/entrypoint.rs [hunk 2]
```diff
@@ -1,3 +1,9 @@
+//! The Rust-based BPF program entry point supported by the latest BPF loader.
```
Comment (CriesofCarrots) [inline sdk/src/entrypoint.rs:1]: I was going to recommend "entrypoint" be one word, but seems like we're not consistent about that. Alas, and nevermind.

File: sdk/src/genesis_config.rs [hunk 3]
```diff
@@ -1,4 +1,4 @@
-//! The `genesis_config` module is a library for generating the chain's genesis config.
+//! Generating the chain's genesis config.
```
Comment (CriesofCarrots) [inline sdk/src/genesis_config.rs:1]:
```suggestion
//! The chain's genesis config.
```
Maybe?

File: sdk/src/hard_forks.rs [hunk 4]
```diff
@@ -1,5 +1,5 @@
-//! The `hard_forks` module is used to maintain the list of slot boundaries for when a hard fork
-//! should occur.
+//! Used to maintain the list of slot boundaries for when a hard fork should
```
Comment (CriesofCarrots) [inline sdk/src/hard_forks.rs:1]:
```suggestion
//! The list of slot boundaries at which a hard fork should
```

File: sdk/src/builtins.rs [hunk 5]
```diff
@@ -1,4 +1,4 @@
-//! Solana builtin helper macros
+//! Solana helper macros for declaring builtin programs.
```
Comment (CriesofCarrots) [inline sdk/src/builtins.rs:1]:
```suggestion
//! Solana helper macros for declaring built-in programs.
```
I don't think we're consistent with this, but this is my preference

File: sdk/src/builtins.rs [hunk 6]
```diff
@@ -58,7 +58,7 @@ macro_rules! declare_builtin_name {
     };
 }
 
-/// Convenience macro to declare a builtin
+/// Convenience macro to declare a builtin.
```
Comment (CriesofCarrots) [inline sdk/src/builtins.rs:61]:
```suggestion
/// Convenience macro to declare a built-in program.
```

File: sdk/src/rpc_port.rs [hunk 7]
```diff
@@ -1,3 +1,5 @@
+//! PRC port numbers.
```
Comment (CriesofCarrots) [inline sdk/src/rpc_port.rs:1]:
```suggestion
//! RPC default port numbers.
```

File: sdk/src/lib.rs [hunk 8]
```diff
@@ -1,3 +1,34 @@
+//! The Solana host and client SDK.
+//!
+//! This is the base library for all off-chain programs that interact with
+//! Solana or otherwise operate on Solana data structures. On-chain programs
+//! instead use the [`solana-program`] crate, the modules of which are
+//! re-exported by this crate, similarly to the relationship between the Rust
```
Comment (CriesofCarrots) [inline sdk/src/lib.rs:6]:
```suggestion
//! re-exported by this crate, like the relationship between the Rust
```

**Review by CriesofCarrots — COMMENTED** [review]
Few thoughts, nothing major

File: sdk/src/entrypoint.rs (hunk 2 above)
Comment (brson) [reply sdk/src/entrypoint.rs:1]: Yeah, this is just copied from the solana-program docs. I'll look at the other docs and see if we can use "entrypoint" everywhere.

File: sdk/src/derivation_path.rs (hunk 1 above)
Comment (brson) [reply sdk/src/derivation_path.rs]: I've modified this to indicate it works with general bip-44 paths.

This module needs more thorough docs, but not now.

File: sdk/src/entrypoint.rs (hunk 2 above)
Comment (brson) [reply sdk/src/entrypoint.rs]: I've changed all the entrypoint docs to spell it with one word.

**Comment by brson** [comment]
Thanks for the good review. All comments addressed.

**Review by CriesofCarrots — APPROVED** [review]
Beautiful. Thank you very much!

**Pull request merged**